src/include/btree_inline.h

```diff
@@ -1612,6 +1612,21 @@ __wt_page_del_visible_all(WT_SESSION_IMPL *session, WT_PAGE_DELETED *page_del, b
     if (page_del == NULL)
         return (true);
 
+    /*
+     * committed flag indicaes whether truncate transaction is committed or not.
+     * In case of prepared transaction, truncate operation is committed in two phases.
+     * In the first phase prepared state is set to resolved and in the later phase the committed flag
+     * of all fast truncated pages is set to true.
+     * The two phase design is to handle any page split cases, when a fast truncated page is
+     * read back and later that page split, in which case all the new split pages will not have page_del info.
+     *
+     * There is a window between setting the prepared state to resolved and setting the committed flag to true.
+     * Hence if committed flag is not set, consider it as not committed irrespective of visibility check, to
+     * avoid checking visibility when transaction commit is in progress.
+     */
+    if (!page_del->committed)
```

Comment:
Should this be a call to `wt_page_del_committed`? It might not be necessary now, but it was subtle enough to warrant a function at some stage, it is probably worth using that function here.

Reply:
Agree and updated. Thank you.